changelog.yml

```diff
@@ -1,4 +1,60 @@
 ---
+- version: 0.19.0
+  date: 2021-03-04
+  summary:
+  added:
+  - |-
+    New `component_dirs` setting on `Dry::System::Container`, which must be used for specifying the directories which dry-system will search for component source files.
+
+      Each added component dir is relative to the container's `root`, and can have its own set of settings configured:
+
+      ```ruby
+      class MyApp::Container < Dry::System::Container
+        configure do |config|
+          config.root = __dir__
+
+          config.component_dirs.add "lib" do |dir|
+            dir.auto_register = true    # defaults to true
+            dir.add_to_load_path = true # defaults to true
+            dir.default_namespace = "my_app"
+          end
+
+          config.component_dirs.add "custom_components" do |dir|
+            # etc.
+          end
+        end
+      end
+      ```
+
+      The following settings are available for configuring added `component_dirs`:
+
+      - `auto_register`, a boolean, or a proc accepting a `Dry::System::Component` instance and returning a truthy or falsey value. Providing a proc allows an auto-registration policy to apply on a per-component basis
+      - `add_to_load_path`, a boolean
+      - `default_namespace`, a string representing the leading namespace segments to be stripped from the component's identifier (given the identifier is derived from the component's fully qualified class name)
+      - `loader`, a custom replacement for the default `Dry::System::Loader` to be used for the component dir
+      - `memoize`, a boolean, to enable/disable memoizing all components in the directory, or a proc accepting a `Dry::System::Component` instance and returning a truthy or falsey value. Providing a proc allows a memoization policy to apply on a per-component basis
+
+      All component dir settings are optional.
+
+      (@timriley in #155 and #157)
+  - |-
+    A new autoloading-friendly `Dry::System::Loader::Autoloading` is available, which is tested and works with [Zeitwerk](https://github.com/fxn/zeitwerk)
+
+    Configure this on the container (either via the container's top-level `loader` setting or as the `loader` for any `component_dirs`), and the loader will no longer `require` any components, instead allowing missing constant resolution to trigger the loading of the required file.
+
+    This loader presumes an autoloading system like Zeitwerk has already been enabled and appropriately configured. (@timriley in #153)
```

Comment:
@timriley I would add a snippet here to show how to configure Zeitwerk. It would be great as selling point: it works with Zeitwerk and here's how you do in a few lines of code.

Reply:
@timriley @jodosha TBH this deserves a blog post and we could link to it from the CHANGELOG. I'm happy to write it/help writing it if needed. This is too good to "just go into a CHANGELOG", if you know what I mean.

Reply:
@solnic @timriley Yeah a blog post is even better. But, as you know, in communication, better to repeat yourself. I'd add an example here, and write a blog post to announce it. 😉